internal/engine/compiler/impl_amd64.go

```diff
@@ -3598,21 +3598,49 @@ func (c *amd64Compiler) compileLoadDataInstanceAddress(dataIndex uint32, dst asm
 	)
 }
 
-// compileMemoryCopy implements compiler.compileMemoryCopy for the amd64 architecture.
-func (c *amd64Compiler) compileMemoryCopy() error {
-	return c.compileCopyImpl(false, 0, 0)
-}
+func (c *amd64Compiler) compileCopyLoopImpl(destinationOffset, sourceOffset, copySize *runtimeValueLocation, backwards bool, bwOffset uint8) {
+	// skip if nothing to copy
+	c.assembler.CompileRegisterToConst(amd64.CMPQ, copySize.register, 0)
+	emptyEightGroupsJump := c.assembler.CompileJump(amd64.JEQ)
 
-// compileCopyImpl implements compileTableCopy and compileMemoryCopy.
-//
-// TODO: the compiled code in this function should be reused and compile at once as
-// the code is independent of any module.
-func (c *amd64Compiler) compileCopyImpl(isTable bool, srcTableIndex, dstTableIndex uint32) error {
-	outOfBoundsErrorStatus := nativeCallStatusCodeMemoryOutOfBounds
-	if isTable {
-		outOfBoundsErrorStatus = nativeCallStatusCodeInvalidTableAccess
+	rvalues := []*runtimeValueLocation{destinationOffset, sourceOffset, copySize}
+
+	// prepare registers for swaps. There will never be more than 3 XCHGs in total
```

Comment:
Please make sure that _all_ of your line comments start with Captial letter and end with period. See other comments. (though trailing comments are not the same)

```suggestion
	// Prepare registers for swaps. There will never be more than 3 XCHGs in total.
```